custom_components/healthchecksio/__init__.py

```diff
@@ -81,15 +86,16 @@ async def update_data(self):
         Logger("custom_components.healthchecksio").debug("Running update")
         # This is where the main logic to update platform data goes.
         try:
-            session = async_get_clientsession(self.hass)
+            verify_ssl = self.site_root.startswith("https")
+            session = async_get_clientsession(self.hass, verify_ssl)
             headers = {"X-Api-Key": self.api_key}
             async with async_timeout.timeout(10, loop=asyncio.get_event_loop()):
                 data = await session.get(
-                    "https://healthchecks.io/api/v1/checks/", headers=headers
+                    f"{self.site_root}/api/v1/checks/", headers=headers
                 )
                 self.hass.data[DOMAIN_DATA]["data"] = await data.json()
 
-                await session.get(f"https://hc-ping.com/{self.check}")
+                await session.get(f"{self.site_root}/{self.ping_endpoint}/{self.check}")
```

Comment:
This will break the hosted option.

Reply:
Well I feared this isnt sufficient, but I thought the HC-ping domain was just a forwarding. I guess we have to differentiate the official and self-hosted domains more specifically then. I will definitely check the official site myself before opening this up for review.

Reply:
```
{self.site_root if "healthchecks.io" not in self.site_root  else "https://hc-ping.com"}
```
Should do the trick

Reply:
Thx for the suggestion. I am currently a bit limited with my time, but will definitely come back to this asap.

Reply:
Just ping me when you are ready :)